File: tests/integration/clients/ping_pong.py
import sys
import os
import time

sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '../../../clients/python/src')))
from nerve.core import NexusClient

client = NexusClient()
client.connect("py_client")

def handle(msg):
    # msg is the full message dict: {"type": ..., "payload": ..., "from": ...}
    payload = msg.get("payload") if isinstance(msg, dict) else None
    if not isinstance(payload, dict):
        return
    if payload.get("event") == "ping":
        client.broadcast({
            "event": "pong",
            "from": "py_client",
            "timestamp": payload.get("timestamp"),
        })

client.listen(handle)

try:
    while True:
        time.sleep(1)
except KeyboardInterrupt:
    pass
finally:
    client.disconnect()

File: tests/integration/clients/ping_pong.js
const { NexusClient } = require('../../../clients/javascript/index.js');

const client = new NexusClient();

client.on('reconnect', (attempt) => {
    console.log(`[JS] Reconnecting... (Attempt ${attempt})`);
});

client.connect('js_client').then(() => {
    client.listen((msg) => {
        // msg is the full message: {"type": ..., "from": ..., "payload": {...}}
        const payload = msg && msg.payload;
        if (!payload) return;

        if (payload.event === 'ping') {
            console.log(`[JS] Received ping from ${msg.from || 'unknown'}, sending pong...`);
            // Echo timestamp back to calculate latency
            client.broadcast({
                event: 'pong',
                from: 'js_client',
                timestamp: payload.timestamp,
            });
        } else if (payload.event === 'pong') {
            console.log(`[JS] Received pong from ${payload.from}`);
        }
    });
}).catch(console.error);

// Keep alive until terminated by the test harness.
setInterval(() => {}, 1000);
